### ISSUE TITLE
Standardize fuel types and remove hidden synonyms

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5941]
Issue overview
--------------
Prompted by the discussion of Propane vs PropaneGas as a fuel type in #5940, searches of the code and IDD show that there are some variations in the list of valid fuel types/resource types in the IDD, input processing functions and the documentation.  This should be cleaned up to a single standard list.

*The following are adressed in #7523.*
1. Standardize on a single list of fuel types
3. Remove all variations that are currently accepted in some parts of the code (e.g. "Gas", "LPG", both "Propane" and "PropaneGas", etc.) - @macumber suggests that these generate warnings for one release then are removed in the next
4. Make transition rules to convert all of these synonyms to their standard name. (This may happen in phase II - see above.)
5. Change the meter resource type names to exactly match the fuel type input names (most of them match now, but Propane, for example does not.
7. Clean up the documentation to make sure every instance that lists fuel types or resource types is consistent.
9. Also, watch out for Generator:FuelSource which has a list of gas components that include "Propane" where this is probably the best name to use next to other choices like Hexane, Methane, etc.

*These will be done separately after #7523. See #7621 and #7622.*

1. Use the same function everywhere to validate fuel type inputs and set the internal type variable - fuel types are a subset of the full list of resource types - be careful here.
2. Be sensitive to exceptions where one or more fuel types might not make sense for a particular object.
3. Review output table headings - currently for example, ABUPS report has columns for "Natural Gas" and "Additional Fuel".

Proposed standard list
-------------------------
Electricity
NaturalGas
Propane (preferred over PropaneGas per discussions in [#5940](https://github.com/NREL/EnergyPlus/pull/5940#issuecomment-270479826)
Diesel
Gasoline
FuelOil#1 (or FuelOil1)
FuelOil#2 (or FuelOil2)
Coal
OtherFuel1
OtherFuel2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by macumber** [comment]
If this is a standardization effort should we get rid of '#' characters?

**Comment by mjwitte** [comment]
@macumber We could.  I think it started that way because fuel oils are commonly referred to as No. 1 Fuel Oil and No. 2 Fuel Oil.  But I don't have a strong opinion on this.  Does look a bit strange to have # in those keys and not in OtherFuel1/2.

**Comment by nmerket** [comment]
I'd prefer using just "Propane" instead of "PropaneGas".

**Comment by Myoldmopar** [comment]
Does #5986 complete this then @mjwitte ?

**Comment by mjwitte** [comment]
No, just a start.  This one addresses #4851 (which I just added to the description at the top).

**Comment by mjwitte** [comment]
Prompted by comments from @hongtz68 on today's iteration call, consulted https://bedes.lbl.gov/bedes-online/resource. The list of resources there includes "Natural Gas", "Propane", "Fuel Oil No 1", "Fuel Oil No 2".  

1. Looking at table outputs, we're currently using "Natural Gas", "Propane", "Fuel Oil #1", "Fuel Oil #2". So the only question there is for fuel oil: Keep as-is, change to "Fuel Oil 1" or change to "Fuel Oil No 1"?

2. For Output:Variable, we use "Boiler Gas Energy" and "Heating Coil FuelOil#2 Energy" so these would all change, lkely to the long form "Boiler Natural Gas Energy" and "Heating Coil Fuel Oil 2 Energy" or "Heating Coil Fuel Oil No 2 Energy".

3. For meters, we use "Gas:Facility" and "FuelOil#2:Facility" so these would all change to the munged form "NaturalGas:Facility" and "FuelOil2" or "FuelOilNo2".

Preferences on the fuel oil? Other comments?  @rraustad @Myoldmopar @shorowit @hongtz68 @JasonGlazer @jasondegraw

**Comment by jasondegraw** [comment]
I'm for using the BEDES terminology as much as possible, so my first preference is to change to "Fuel Oil No 1", etc.

**Comment by shorowit** [comment]
@mjwitte Thanks for cleaning all of this up! I have no strong opinion on naming convention.

We use a lot of different fuel types in our residential models, and the one thing that has always bothered me is that the End Uses (and similar) output tables lump a bunch of different fuels (fuel oil, propane, otherfuel1, otherfuel2) into "Additional Fuel". This means I often have to do more work to add a bunch of output vars to get out the information I want. Is there any possibility of separating all of the different fuels in these tables? (If we're worried about too many columns, maybe columns that have all zeroes could be automatically hidden?)

**Comment by mjwitte** [comment]
@shorowit Understood on the "Additional Fuel" column. Please post a new issue for that. And, I probably shouldn't ask this. Are there any resource/fuel types that should be added?

**Comment by shorowit** [comment]
Ok, I created a separate issue for the "Additional Fuel" column: https://github.com/NREL/EnergyPlus/issues/7591

Missing fuel types that we've come across are 1) cord wood, 2) wood pellets, and 3) kerosene. It would certainly be nice to allow these, we currently have to resort to OtherFuel1 and OtherFuel2.
